ui/src/user/components/LoginPage/LoginPage.jsx

```diff
@@ -8,13 +8,17 @@ import './LoginPage.scss';
 import ExternalLink from '../../../common/components/ExternalLink';
 import DocumentHead from '../../../common/components/DocumentHead';
 import { WHAT_IS_ORCID_URL } from '../../../common/constants';
+import { USER_SIGNUP } from '../../../common/routes';
 
 const META_DESCRIPTION = 'Log in to your INSPIRE account. Log in with ORCID';
 const TITLE = 'Login';
 
 class LoginPage extends Component {
   render() {
-    const { onLoginClick } = this.props;
+    const { previousUrl } = this.props;
+    const loginHref = `/api/accounts/login?next=${encodeURIComponent(
```

Comment:
`previousUrl` should already be uri friendly no?

Reply:
Yeah actually it's useless, I changed it.